rasa/nlu/utils/spacy_utils.py

```diff
@@ -88,21 +86,71 @@ def cache_key(
         cls, component_meta: Dict[Text, Any], model_metadata: "Metadata"
     ) -> Optional[Text]:
 
-        # Fallback, use the language name, e.g. "en",
-        # as the model name if no explicit name is defined
-        spacy_model_name = component_meta.get("model", model_metadata.language)
+        spacy_model_name = cls._check_model_fallback(
+            component_meta.get("model"), model_metadata.language, warn=False
+        )
 
         return cls.name + "-" + spacy_model_name
 
+    @staticmethod
+    def _check_model_fallback(
+        spacy_model_name: Union[str, None], language_name: str, warn: bool = False
+    ):
+        """This method checks if the `spacy_model_name` is missing.
+
+        If it is missing, we will attempt a fallback. This feature is a measure
+        to support spaCy 3.0 without breaking on users. In the future
+        spaCy will no longer support `spacy link`.
+        """
+        if not spacy_model_name:
+            fallback_mapping = {
+                "zh": "zh_core_web_md",
+                "da": "da_core_news_md",
+                "nl": "nl_core_news_md",
+                "en": "en_core_web_md",
+                "fr": "fr_core_news_md",
+                "de": "de_core_news_sm",
+                "el": "el_core_news_md",
+                "it": "it_core_news_md",
+                "ja": "ja_core_news_md",
+                "lt": "lt_core_news_md",
+                "mk": "mk_core_news_md",
+                "nb": "nb_core_news_md",
+                "pl": "pl_core_news_md",
+                "pt": "pt_core_news_md",
+                "ro": "ro_core_news_md",
+                "ru": "ru_core_news_md",
+                "es": "es_core_news_md",
+            }
+            if language_name not in fallback_mapping.keys():
+                raise InvalidModelError(
+                    f"There is no fallback model for language '{language_name}'."
+                    f" Please add a `model` property to `SpacyNLP` manually to prevent this."
+                    f"More informaton can be found on {DOCS_URL_COMPONENTS}#spacynlp"
```

Comment:
```suggestion
                    f"There is no fallback model for language '{language_name}'. "
                    f"Please add a `model` property to `SpacyNLP` manually to prevent this. "
                    f"More informaton can be found on {DOCS_URL_COMPONENTS}#spacynlp"
```